diff --git a/ykman/otp.py b/ykman/otp.py
index f4e2897..33b5d78 100644
--- a/ykman/otp.py
+++ b/ykman/otp.py
@@ -35,6 +35,7 @@ from http.client import HTTPSConnection
 from datetime import datetime
 from typing import Iterable, Optional

+import ssl
 import json
 import struct
 import random
@@ -106,8 +107,15 @@ def _prepare_upload_key(
         "public_id": modhex_public_id,
         "private_id": private_id.hex(),
     }
-
-    httpconn = HTTPSConnection(_UPLOAD_HOST, timeout=1)  # nosec
+    try:
+        from pip._vendor import certifi
+
+        context = ssl.SSLContext(ssl.PROTOCOL_TLS_CLIENT)
+        context.load_verify_locations(certifi.where())
+    except Exception as e:
+        logger.error("Unable to load certifi CA bundle.", exc_info=e)
+        context = ssl.create_default_context()
+    httpconn = HTTPSConnection(_UPLOAD_HOST, timeout=1, context=context)  # nosec

     try:
         httpconn.request(